model/clusters_mgmt/v1/cluster_operators_info_type.model

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright (c) 2019 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Provides detailed information about the Metrics of the Cluster
+struct OperatorsInfo {
```

Comment:
Rename to "ClusterOperatorsInfo".

@cben I would like to understand what is the reason for using this "operators" attribute. Do we expect to have other attributes in this type? Can't we use an array directly:

```
GET /api/clusters_mgmt/v1/clusters/123/metric_queries/cluster_operators

[
  {
    "time": "...",
    "name": "...",
    "reason": "...",
    "version": "...",
    "condition": "..."
  },
  {  
    "time": "...",
    "name": "...",
    "reason": "...",
    "version": "...",
    "condition": "..."
  },
  ...
]
```

Or even better, to be more consistent with the rest of the API, use paging?

Reply:
Attribute vs top-level array: to be honest I don't remember.  What I wrote in [original design doc](https://docs.google.com/document/d/1XQiMfO0p1QvPoz-MHqUgR1NQgaQRAA2EPt0_kMjFIkI/edit?ts=5d2744ae#heading=h.h67er2n89p9m) was maybe FUD:

> Top level of successful JSON API response should be an object (even if we all have is a single number), and have known key(s), none of which is "error".  This is simplest for compatibility (RFC 7159 naked value not universally allowed yet I think?), and more importantly allows UI to easily distinguish error responses from data.

is more relevant for queries that return single number, but top-level array is supported since the original JSON RFC 4627.  But "error" case is a good reason IMHO.

Reply:
I think also I intially designed it as hash with operator names as keys, and converted to array per feedback from Liza that array would be convenient for UI.

Reply:
Paging/collection: 
- We originally discussed it for the whole `metric_queries/` endpoint, where it would make sense for discovery of supported queries.  I didn't want it because it's extra work; IIRC we never quite agreed on this, and my lazy approach only won because our goal at the time was to make queries _as easy as possible_ to implement :smile:

- I don't remember us considering paging for data *inside* a query result.  
  Hmm, for cluster operators it *could* be a consistent API to expose.  
  IMHO it's again not worth it because it's extra work to implement in CS, useless for performance because CS would still get full results from prometheus and drop parts (at least I don't see good OFFSET/LIMIT-like functionality in PromQL), and not useful for UI which just wants them all.

  We also have other array use cases for grouping samples over time, e.g. `metric_queries/cpu_usage`, where modeling as a pagable collection is weirder IMO.  If fetching a sub-range were needed, specifying start/end times is better than page/size.  Anyway those queries are not used at all (requirements changed after we implemented them).

Reply:
[some context for Priya: we've had recurring arguments on the team about applying our [API guidelines](https://gitlab.cee.redhat.com/service/api-guidelines) collection pattern with search & paging everywhere vs selectively...  
While we have reasons for both sides, time proved it's also matter of subjective taste and not very productive :wink:]

@jhernand if you decide we should change the api, please open a ticket :+1: